crates/burn-ndarray/src/ops/tensor.rs

```diff
@@ -338,6 +338,17 @@ impl<E: FloatNdArrayElement> FloatTensorOps<Self> for NdArray<E> {
         NdArrayMathOps::sum_dim(tensor, dim)
     }
 
+    fn float_cumsum_dim<const D: usize>(
+        tensor: NdArrayTensor<E, D>,
+        dim: usize,
+    ) -> NdArrayTensor<E, D> {
+        let mut array = tensor.array.clone().into_owned();
```

Comment:
I *believe* the underlying array struct of `tensor` needs to be cloned, since NdArray's method for accumulating elements along an axis modifies an array's data inplace.  Referring to [this method](https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html#method.accumulate_axis_inplace)

Reply:
Well `float_cumsum` takes ownership of the `tensor`, so I don't think the clone is required here.